gobblin-runtime/src/main/java/org/apache/gobblin/runtime/job_monitor/AvroJobSpecKafkaJobMonitor.java

```diff
@@ -108,10 +102,10 @@ protected void createMetrics() {
   /**
    * Creates a {@link JobSpec} or {@link URI} from the {@link AvroJobSpec} record.
    * @param record the record as an {@link AvroJobSpec}
-   * @return a {@link JobSpec} or {@link URI} wrapped in a {@link Collection} of {@link Either}
+   * @return a {@link JobSpec}
    */
   @Override
-  public Collection<Either<JobSpec, URI>> parseJobSpec(AvroJobSpec record) {
+  public Collection<JobSpec> parseJobSpec(AvroJobSpec record) {
```

Comment:
A failure in maintaining backward compatibility ?

Reply:
In KafkaJobMonitor, caller of this api is supposed to decide whether it is an insert/update or delete type of job spec based on Either c. I could not found any other way to represent a 3rd type of job spec. It also makes less sense to me, because the job spec itself has sufficient information for caller to decide what kind of request it is; jobspec's metadata has a field VERB to store this information.
Yes it is a backward incompatible change, I changed all the implementations of KafkaJobMonitor using this new return type, including this one (AvroJobSpecKafkaJobMonitor). If there still is any implementation of KafkaJobMonitor by external developers, probably this change can help them how to remove either?